nmmo/core/env.py

```diff
@@ -186,16 +154,23 @@ def reset(self, map_id=None, seed=None, options=None):
       if isinstance(ent.agent, Scripted):
         self.scripted_agents.add(eid)
 
-    self.tasks = copy.deepcopy(self._tasks)
     self.obs = self._compute_observations()
     self._gamestate_generator = GameStateGenerator(self.realm, self.config)
 
-    gym_obs = {}
-    for a, o in self.obs.items():
-      gym_obs[a] = o.to_gym()
-      if self._task_encoding:
-        gym_obs[a]['Task'] = self._encode_goal().get(a,np.zeros(self._task_embedding_size))
-    return gym_obs
+    """Two methods to define tasks.
+         * new_tasks: a list of instantiated tasks. This method has precedence
+         * task_spec and teams: these are used to instantiate tasks here
+       If these are all None, then use the default task
+    """
+    if new_tasks is not None:
+      # providing an empty new_tasks [] is also possible
+      self.tasks = new_tasks
+    elif task_spec is not None and teams is not None:
+      self.tasks = make_team_tasks(teams, task_spec)
+    else:
+      self.tasks = nmmo_default_task(self.possible_agents)
```

Comment:
this is going to replace existing tasks with the default_task if we don't pass in new_tasks, which is not expected. how about we set self.tasks to default_task in the constructor, and then don't do that here

Reply:
nmmo_default_task has moved to the __init__, and this line was replaced with
```
      for task in self.tasks:
        task.reset()
```
not changing the existing tasks.